cms/functions.php

```diff
@@ -0,0 +1,260 @@
+ * @param $projectImage string object used with post from form and passed in to update the portfolio table in database.
+ * @param $projectId string used with post from form and passed in to update the portfolio table in database.
+ *
+ * @return array returns associative array with arrays containing content from portfolio table from database.
+ */
+function updateProject(PDO $db, string $projectTitle, string $projectTitleLink, string $projectImage, string $projectId) {
+
+    $query = $db->prepare("UPDATE `portfolio` SET `projectTitle` = :projectTitle, `projectImage` = :projectImage, `projectTitleLink` = :projectTitleLink WHERE `id` = :projectId");
+
+    $query->bindParam (':projectTitle',$projectTitle);
+    $query->bindParam (':projectTitleLink',$projectTitleLink);
+    $query->bindParam (':projectImage',$projectImage);
+    $query->bindParam (':projectId',$projectId);
+
+    $query->execute();
+
+    $updatedInfo = $projectId.' '. $projectTitle.' '.$projectTitleLink.' '.$projectImage;
+    return $updatedInfo;
+}
+
+/* Doc Block
+ * Gets user credentials from the database.
+ *
+ * @param $db object database link and credentials PDO object.
+
+ *
+ * @return array returns associative array with arrays credentials from users and passwords tables from database.
+ */
+function getUserCredentials(PDO $db):array {
+    $userCredentialsQuery = $db->prepare("SELECT `users`.`name`, `password` FROM `users` LEFT JOIN `passwords` ON `users`.`id` = `passwords`.`userId`");
+    $userCredentialsQuery->execute();
+    return $userCredentials = $userCredentialsQuery->fetchAll();
+}
+
+/* Doc Block
+ * Checks login details are validated and match saved credentials.
+ *
+ * @param $actualUserName string username stored credential.
+ * @param $actualPassword string password stored credential.
+ * @param $inputUserName string form inout username.
+ * @param $inputPassword string form input password.
+ *
+ * @return bool returns true if both content types are string.
+ */
+function checkCredentials(string $actualUserName, string $actualPassword, string $inputUserName, string $inputPassword): bool {
+
+
+    if (($inputUserName === $actualUserName) && ($inputPassword === $actualPassword)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/* Doc Block
+ * Directs to admin page or echoes instructions depending on credentials check.
+ *
+ *  @param $actualUserName string username stored credential.
+ * @param $actualPassword string password stored credential.
+ * @param $inputUserName string form inout username.
+ * @param $inputPassword string form input password.
+ *
+ * @return bool returns true if both content types are string.
+ */
+
+
+function loggedInSession(string $actualUserName, string $actualPassword, string $inputUserName, string $inputPassword ):bool {
+    $loginCheck = checkCredentials($actualUserName, $actualPassword, $inputUserName, $inputPassword);
```

Comment:
best practice not to call functions inside other functions. abstract out and pass in result of function call into this function